### PR TITLE
Regex multiple matches 

### PR DESCRIPTION
**Pull request by mouhamadalmounayar**
### Types of changes

- [ ] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

- The extractMatching function has been enhanced to support multiple matches using regular expressions. Previously, the function returned the first single match within the input parameter. The updated function now allows for capturing all matches if the `/g` flag is used, and concatenating them into a single string separated by spaces.

### CODE REVIEW

**Comment by jmcook1186** [comment]
looks good, thanks @mouhamadalmounayar - will get this reviewed shortly.
In order to merge, we need signed commits. You can satisfy the DCO requirement by adding a DCO remediation commit, as described here: https://github.com/Green-Software-Foundation/if/pull/901/checks?check_run_id=27267934777

File: src/if-run/builtins/regex/README.md [hunk 1]
```diff
@@ -22,7 +22,8 @@ Intel® Xeon® Platinum 8272CL,Intel® Xeon® 8171M 2.1 GHz,Intel® Xeon® E5-26
 
 ## Returns
 
-- `output`: the first match of `parameter` with the parameter name with `match` defined in global config.
+- `output`: The first match of the `parameter` value using the `match` regex defined in the global config. If the `match` regex includes the global flag (`g`), a string containing all matches seperated by spaces.
```
Comment (manushak) [inline src/if-run/builtins/regex/README.md:25]:
```suggestion
- `output`: The match of the `parameter` value using the `match` regex defined in the global config. If the `match` regex includes the global flag (`g`), a string containing all matches separated by spaces.
```

**Review by manushak — APPROVED** [review]

**Comment by jmcook1186** [comment]
Thanks for contributing @mouhamadalmounayar :o)

**Review by jmcook1186 — APPROVED** [review]

**Comment by jmcook1186** [comment]
@mouhamadalmounayar looks like we haven't quite satisfied the DCO - maybe there was an error in your remediation commit. You can either squash, sign and force push or try again to add a remediation commit that covers the whole commit history.

**Comment by mouhamadalmounayar** [comment]
> @mouhamadalmounayar looks like we haven't quite satisfied the DCO - maybe there was an error in your remediation commit. You can either squash, sign and force push or try again to add a remediation commit that covers the whole commit history.

Yeah sorry about that, I had a problem with my gpg config. All good now.

**Pull request merged**